test/NuGet.Core.FuncTests/NuGet.Packaging.FuncTest/TimestampProviderTests.cs

```diff
@@ -0,0 +1,201 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Security.Cryptography.Pkcs;
+using System.Text;
+using System.Threading;
+using FluentAssertions;
+using NuGet.Packaging.Signing;
+using NuGet.Test.Utility;
+using Org.BouncyCastle.X509;
+using Test.Utility.Signing;
+using Xunit;
+
+namespace NuGet.Packaging.FuncTest
+{
+    public class TimestampProviderTests
+    {
+        private const string _internalTimestamper = "http://rfc3161.gtm.corp.microsoft.com/TSS/HttpTspServer";
+        private const string _authorCertExpiredExceptionMessage = "Author certificate was not valid when it was timestamped.";
+        private const string _argumentNullExceptionMessage = "Value cannot be null.\r\nParameter name: {0}";
+        private const string _operationCancelledExceptionMessage = "The operation was canceled.";
+
+        [Fact]
```

Comment:
Use a conditional fact so that this only runs on the CI. Other func tests for protocol have this.

For external users working on PRs, if any tests fail they tend to give up or ignore failures caused by their changes.